mycroft/client/speech/mic.py

```diff
@@ -359,6 +360,8 @@ def __init__(self, loop, watchdog=None):
         # Signal statuses
         self._stop_signaled = False
         self._listen_triggered = False
+        self._waiting_for_wakeup = False
```

Comment:
Should this be set to loop.state.sleeping?

Reply:
no, this is a flag for the wake word detection when in sleep mode, it should only be set to True when "hey mycroft" is heard in sleep mode. This basically starts a 5 second time window when "wake up" is a valid detection

Reply:
> I'm probably not seeing something, but in `mic.py`, is `self._waiting_for_wakeup` not always in the same state as `loop.state.sleeping`?

you enter in sleep mode and self._waiting_for_wakeup is False, it only becomes True (for 5 seconds) once a valid wake word is heard, after 5 seconds goes back to False.